server.js

```diff
@@ -0,0 +1,123 @@
+var Hapi = require('hapi');
+// var Path =
+var Inert = require('inert');
+var Vision = require('vision');
+var Handlebars = require('handlebars')
+
+var Request = require('request');
+
+var server = new Hapi.Server();
+
+server.connection({
+    port: 8000,
+    host: 'localhost'
+});
+
+server.register(Inert, function(err) {
+    if (err) {
+        throw err;
+    }
+
+    server.route({
+        method: "GET",
+        path: '/public/{path*}',
+        handler: {
+            directory: {
+                path: "./public",
+                listing: false,
+                index: false
+            }
+        }
+    });
+});
+
+server.register(Vision, function(err) {
+    if (err) {
+        throw err;
+    }
+
+    //configure template support
+    server.views({
+        engines: {
+            html: Handlebars
+        },
+        path: './public',
+        //need to figure out how to render partials and use default layout
+        // layoutPath: Path.join(__dirname, 'views/layouts')
+    })
+});
+
+server.route({
+    method: 'GET',
+    path: '/',
+    handler: function(request, reply) {
+        reply.view('index');
+    }
+});
+
+
+
+
+server.route({
+            method: 'POST',
+            path: '/starter-issue',
+            handler: function(request, reply) {
+
+                //url user submits
+                var url = request.payload.url;
+                console.log("This is the URL submitted: " + url)
+                var secondURL = url.replace('https://github.com', 'https://api.github.com/repos');
+                var finalURL = secondURL.replace('tree', 'branches');
+                // console.log(finalURL) //https://api.github.com/repos/Techforchange/doctoc/branches/update
+
+                var branchRequest = {
+                    url: finalURL,
+                    headers: {
+                        'User-Agent': 'request'
+                    }
+                };
+
+                //branch information
+                function callback(err, result, body) {
+                    var parsedBody = JSON.parse(result.body);
+                    var sha = parsedBody.commit.sha;
+
+                    console.log("The SHA: " + sha);
+
+                    var commitURL = finalURL.replace('branches/update', 'commits/' + sha);
```

Comment:
This only works if the URL you submitted is for a branch called `"update"`. So you have to make it a bit more flexible. I’m not sure if you worked with regular expressions before, but that is exactly what we will need here. Try this instead: 

```js
var commitURL = finalURL.replace(/branches.*$/, 'commits/' + sha);
```

This says find everything that starts with "branches" followed by all characters until the end of the string, and replace that with `'commits/' + sha`